src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -2467,4 +2491,45 @@ private function initConnection()
 
         $this->connectionInitialized = true;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function checkinItem($path)
+    {
+        return $this->getVersionHandler()->checkinItem($path);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function checkoutItem($path)
+    {
+        return $this->getVersionHandler()->checkoutItem($path);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function restoreItem($removeExisting, $versionPath, $path)
+    {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function removeVersion($versionPath, $versionName)
+    {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * Sets the generic version handler delivered by jackalope
+     * @param VersionHandler $versionHandler
+     */
+    public function setVersionHandler(VersionHandler $versionHandler)
+    {
+        $this->versionHandler = $versionHandler;
```

Comment:
this is a classic "optional dependency". i think there is no way to avoid it here. but lets at least add a check `if ($this->versionHandler) { throw new \InvalidArgumentException('Version handler is already set'); }`